caom2utils/caom2utils/tests/test_data_util.py

```diff
@@ -107,62 +107,64 @@ def test_get_file_type():
 def test_storage_inventory_client(cadc_client_mock):
     test_subject = Mock(autospec=True)
     test_uri = 'cadc:TEST/test_file.fits'
-    test_working_directory = Path(test_fits2caom2.TESTDATA_DIR)
-    test_fqn = test_working_directory / 'test_file.fits'
-    if test_fqn.exists():
-        test_fqn.unlink()
-
-    def info_si_mock(ignore):
-        return FileInfo(id=test_uri, file_type='application/fits', md5sum='abc', size=42)
-
-    def get_si_mock(ignore2, dest, **kwargs):
-        fhead = kwargs.get('fhead')
-        if fhead:
-            dest.write(TEST_HEADERS)
-        else:
-            test_fqn.write_text('StorageInventoryClient')
-
-    cadc_client_mock.return_value.cadcinfo.side_effect = info_si_mock
-    cadc_client_mock.return_value.cadcget.side_effect = get_si_mock
-    cadc_client_mock.return_value.cadcput = Mock(autospec=True)
-    cadc_client_mock.return_value.cadcremove = Mock(autospec=True)
-
-    test_wrapper = data_util.StorageClientWrapper(subject=test_subject)
-    assert test_wrapper is not None, 'ctor failure'
-
-    # info
-    test_result = test_wrapper.info(test_uri)
-    _check_info_result(test_result)
-
-    # get_head
-    test_result = test_wrapper.get_head(test_uri)
-    _check_header_result(test_result)
+    for test_working_directory in [Path(test_fits2caom2.TESTDATA_DIR), Path('./')]:
+        test_fqn = test_working_directory / 'test_file.fits'
+        if test_fqn.exists():
+            test_fqn.unlink()
+
+        def info_si_mock(ignore):
```

Comment:
I'm not sure what the implications (if any) of having the function definition inside a for loop..

Reply:
I moved the function definitions outside of the loop.